src/clients/TokenClient.ts

```diff
@@ -406,6 +411,74 @@ class TokenClient {
     return this.tokens;
   }
 
+  public async getBridges(): Promise<TypeUtils.SimpleMap<Bridge[]>> {
+    const allBridges = await this.query.coin.BridgeAll({
+      pagination: {
+        key: new Uint8Array(),
+        limit: new Long(100),
+        offset: Long.UZERO,
+        countTotal: true,
+        reverse: false,
+      },
+    })
+    const ibcBridges = allBridges.bridges.filter(bridge => {
+      if (!bridge.enabled) return
+      return bridge.bridgeId.toNumber() === BRIDGE_IDS.ibc
+    })
+    const polynetworkBridges = allBridges.bridges.filter(bridge => {
+      if (!bridge.enabled) return
+      return bridge.bridgeId.toNumber() === BRIDGE_IDS.polynetwork
+    })
+    Object.assign(this.bridges, {"ibc": ibcBridges, "polynetwork": polynetworkBridges})
+    return this.bridges
+  }
+
+  public getIbcBlockchainNames(): string[] {
+    return this.bridges["ibc"].map(bridge => bridge.chainName)
+  }
+
+  public getPolynetworkBlockchainNames(): string[] {
+    return this.bridges["polynetwork"].map(bridge => bridge.chainName)
+  }
+
+  public getAllBlockchainNames(): string[] {
+    return this.getIbcBlockchainNames().concat(this.getPolynetworkBlockchainNames())
+  }
+
+  public getIbcTokens(): TypeUtils.SimpleMap<Token> {
+    Object.values(this.tokens).forEach(token => {
+      if (token.bridgeId.toNumber() === BRIDGE_IDS.ibc) {
+        this.ibcTokens[token.denom] = token
+      }
+    })
+    return this.ibcTokens
+  }
```

Comment:
Notice you're resetting this.ibcTokens every time you execute this function. Instead of doing that, you can just filter every time you run this function

i.e.
```
const ibcTokens = Object.values(this.tokens).reduce((prev: TypeUtils.SimpleMap<Token>, token: Token) => {
  const newPrev = prev;
  if (token.bridgeId.toNumber() === BRIDGE_IDS.ibc) {
    newPrev[token.denom] = token;
  }
  return newPrev;
}, {})
return this.ibcTokens 
```